server.go

```diff
@@ -702,11 +702,16 @@ func (s *Server) waitForReady() error {
 	// this part of the code, the called component will handle the quit
 	// signal.
 	select {
-	case <-s.ready:
-		return nil
-
 	case <-s.quit:
 		return fmt.Errorf("tapd is shutting down")
+
+	default:
+		select {
+		case <-s.ready:
+			return nil
+		case <-s.quit:
```

Comment:
I think this doesn't solve the `tapd: don't set initialized=true when startup (StartAsSubserver) fails` part of the issue. We don't close the `quit` channel when we fail to start there (which is the main entrypoint for `litd`).

Reply:
Ah right, someone waiting at the `waitForReady` endpoint will stay hanging

Will enhance the `StartAsSubserver` method to conditionally close the `s.quit` chan, if initialization faced any err